pygfx/renderers/wgpu/_pipelinebuilder.py

```diff
@@ -0,0 +1,520 @@
+            pipeline_resources.append(("buffer", buffer))
+        for key in pipeline_info.keys():
+            if key.startswith("bindings"):
+                resources = pipeline_info[key]
+                if isinstance(resources, dict):
+                    resources = resources.values()
+                for binding in resources:
+                    resource = binding.resource
+                    if binding.type.startswith("buffer/"):
+                        assert isinstance(resource, Buffer)
+                        pipeline_resources.append(("buffer", resource))
+                        if "uniform" in binding.type:
+                            resource._wgpu_usage |= wgpu.BufferUsage.UNIFORM
+                        elif "storage" in binding.type:
+                            resource._wgpu_usage |= wgpu.BufferUsage.STORAGE
+                    elif binding.type.startswith("sampler/"):
+                        assert isinstance(resource, TextureView)
+                        pipeline_resources.append(("sampler", resource))
+                    elif binding.type.startswith("texture/"):
+                        assert isinstance(resource, TextureView)
+                        resource.texture._wgpu_usage |= (
+                            wgpu.TextureUsage.TEXTURE_BINDING
+                        )
+                        pipeline_resources.append(("texture", resource.texture))
+                        pipeline_resources.append(("texture_view", resource))
+                    elif binding.type.startswith("storage_texture/"):
+                        assert isinstance(resource, TextureView)
+                        resource.texture._wgpu_usage |= (
+                            wgpu.TextureUsage.STORAGE_BINDING
+                        )
+                        pipeline_resources.append(("texture", resource.texture))
+                        pipeline_resources.append(("texture_view", resource))
+                    else:
+                        raise RuntimeError(
+                            f"Unknown resource binding {binding.name} of type {binding.type}"
+                        )
+
+    return pipeline_resources
+
+
+def create_pipeline_objects(shared, blender, wobject):
+    """Generate wgpu pipeline objects from the list of pipeline info dicts."""
+
+    # Prepare the three kinds of pipelines that we can get
+    compute_pipelines = []
+    render_pipelines = []
+    alt_render_pipelines = []
+
+    # Process each pipeline info object, converting each to a more concrete dict
+    for pipeline_info in wobject._wgpu_pipeline_infos:
+        if "vertex_shader" in pipeline_info and "fragment_shader" in pipeline_info:
+            pipeline = compose_render_pipeline(shared, blender, wobject, pipeline_info)
+            if pipeline_info.get("target", None) is None:
+                render_pipelines.append(pipeline)
+            else:
+                raise NotImplementedError("Alternative render pipelines")
```

Comment:
This should not be here, right?

Reply:
The idea is that material renderers can chose to render to an alternative render target (multi-pass rendering). We don't need it (yet) so it has not been implemented.

Reply:
This code was already there, BTW. It just moved.

Reply:
Thanks for clarifying!